### DISCUSSION TITLE
Audio as input (voice changer / voice conversion)

### DISCUSSION THREAD

**Discussion by etlweather** [myshell-ai/OpenVoice#298] [General]
I was under the impression I could do voice changing (aka voice conversion) with this tool. But I can't find how. Is it possible? I would want to take an audio in and change the voice to another.

**Comment by avzh1** [comment]
Hi,

In their demo_part3 which I was following, you may notice that in the last cell of the notebook the pipeline for generating the text for each language happens by first running a TextToSpeech model with `melo.api`.

It is possible to do an audio to audio conversion by using

```
# Run the tone color converter
encode_message = "@MyShell"
tone_color_converter.convert(
    audio_src_path=f'/your/path/input_audio.wav', 
    src_se=torch.load(f'checkpoints_v2/base_speakers/ses/en-default.pth', map_location=device), 
    tgt_se=target_se, # as before
    output_path=f'{output_dir}/output_v2_default.wav', 
    message=encode_message)
```

Although, note, it might sound dodgy because it seems that the model was trained on this TTS - so in order for it to sound convincing you'd need to identify a method to convert your input_audio.wav into a similar sound as the TTS would...

**Comment by loscrossos** [comment]
yes its possible. dont remember where but there is a code snippet where you take 2 audio files and clone onto the other.

you have to initalize the src_se and target_se to each file and thats it.
